### PR TITLE
Refactor front end config load to resist non present configs

### LINKED ISSUE #6101: box configuration should be optional
If you don't have credentials for 'box' cartodb fails at startup in this line

```
 oauth_box:                  Cartodb.config[:oauth]['box']['client_id'],
```

### PR DESCRIPTION
**Pull request by gfiorav**
Fixes #6101

CR please, @ethervoid

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -74,23 +74,36 @@ def frontend_config
...
       account_host:               CartoDB.account_host,
       dropbox_api_key:            Cartodb.config[:dropbox_api_key],
-      gdrive_api_key:             Cartodb.config[:gdrive]['api_key'],
-      gdrive_app_id:              Cartodb.config[:gdrive]['app_id'],
-      oauth_dropbox:              Cartodb.config[:oauth]['dropbox']['app_key'],
-      oauth_box:                  Cartodb.config[:oauth]['box']['client_id'],
-      oauth_gdrive:               Cartodb.config[:oauth]['gdrive']['client_id'],
-      oauth_instagram:            Cartodb.config[:oauth]['instagram']['app_key'],
-      oauth_mailchimp:            Cartodb.config[:oauth]['mailchimp']['app_key'],
+      gdrive_api_key:             Cartodb.config[:gdrive].present? ? Cartodb.config[:gdrive]['api_key'] : '',
```
Comment (ethervoid) [inline app/helpers/application_helper.rb]: Maybe we could add a method to get a key from the config with default value in [01_app_config.rb](https://github.com/CartoDB/cartodb/blob/master/config/initializers/01_app_config.rb)

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -74,23 +74,36 @@ def frontend_config
...
+      gdrive_app_id:              Cartodb.config[:gdrive].present? ? Cartodb.config[:gdrive]['app_id'] : '',
       bitly_login:                Cartodb.config[:bitly].present? ? Cartodb.config[:bitly]['login'] : '',
       bitly_key:                  Cartodb.config[:bitly].present? ? Cartodb.config[:bitly]['key'] : '',
-      datasource_search_twitter:  nil,
-      tumblr_api_key:             Cartodb.config[:tumblr]['api_key'],
-      max_asset_file_size:        Cartodb.config[:assets]["max_file_size"],
+      tumblr_api_key:             Cartodb.config[:tumblr].present? ? Cartodb.config[:tumblr]['api_key'] : '',
+      max_asset_file_size:        Cartodb.config[:assets].present? ? Cartodb.config[:assets]["max_file_size"] : '',
       watcher_ttl:                Cartodb.config[:watcher].try("fetch", 'ttl', 60),
```
Comment (ethervoid) [inline app/helpers/application_helper.rb]: Maybe you could use try too instead of create a method to retrieve a key

**Comment by ethervoid** [comment]
Seems ok to me despite the two comments, that give robustness to the code. Also maybe create an unitary test for the frontend_config method could be great to avoid problems with the config.

**Comment by gfiorav** [comment]
Thanks, @ethervoid ! It turns out that method `get_config` in `01_app_config.rb` is very useful, so I refactored it to use that method. Default if key isn't present is `nil` which is fine for us as fronted javascript check `if(!object)` will return `true` if object is either `null` or `''`. See if you're ok with this :)

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (gfiorav) [inline app/helpers/application_helper.rb]: I think `max_asset_file_size` is compulsory so i'd leave it. The code is now refactored to do the same with `tumblr_api_key`, though :)

**Comment by ethervoid** [comment]
I was looking for that method hahaha. :+1:

**Comment by gfiorav** [comment]
haha! Ok, thanks @ethervoid ! I'll merge as soon as CI pass :)

**Pull request merged**